test/src/responder/multiResponder.test.ts

```diff
@@ -301,12 +310,137 @@ describe("MultiResponder", () => {
+        await responder.startResponse(appointmentId, responseData);
+        await responder.startResponse(appointmentId2, responseData2);
+
+        const item = responder.respondedTransactions.get(appointmentId)!.queueItem;
+        await responder.txMined(item.request.identifier, item.nonce);
+
+        const queueBefore = responder.queue;
+        await responder.reEnqueueMissingItems([appointmentId, appointmentId2]);
+        const replacedTransactions = responder.queue.difference(queueBefore);
+        expect(replacedTransactions.length).to.equal(1);
+        expect(replacedTransactions[0].request.identifier).to.equal(item.request.identifier);
+        expect(replacedTransactions[0].nonce).to.equal(item.nonce);
+
+        await responder.stop();
+    });
+
+    it("reEnqueueMissingItems does replace transactions", async () => {
+        const appointmentId = "app1";
+        // choose a lower gas fee for the first item - this should cause a double replacement
+        const responseData = createResponseData("app1");
+        const appointmentId2 = "app2";
+        const responseData2 = createResponseData("app2");
+
+        const responder = new MultiResponder(
+            signer,
+            increasingGasPriceEstimator,
+            maxConcurrentResponses,
+            replacementRate
+        );
+        await responder.start();
+        await responder.startResponse(appointmentId, responseData);
+        const item = responder.respondedTransactions.get(appointmentId)!.queueItem;
+        await responder.txMined(item.request.identifier, item.nonce);
+
+        await responder.startResponse(appointmentId2, responseData2);
+        const item2 = responder.respondedTransactions.get(appointmentId2)!.queueItem;
+
+        // should only be one item in the queue
+        expect(responder.queue.queueItems.length).to.equal(1);
+
+        const queueBefore = responder.queue;
+        await responder.reEnqueueMissingItems([appointmentId, appointmentId2]);
+        const replacedTransactions = responder.queue.difference(queueBefore);
+        expect(replacedTransactions.length).to.equal(2);
+        expect(replacedTransactions[0].request.identifier).to.equal(item2.request.identifier);
+        expect(replacedTransactions[0].nonce).to.equal(item.nonce);
+        expect(replacedTransactions[1].request.identifier).to.equal(item.request.identifier);
+        expect(replacedTransactions[1].nonce).to.equal(item2.nonce);
+
+        await responder.stop();
+    });
+
+    it("reEnqueueMissingItems throws error for missing transactions", async () => {
+        const appointmentId = "app1";
+        const responder = new MultiResponder(
+            signer,
+            decreasingGasPriceEstimator,
+            maxConcurrentResponses,
+            replacementRate
+        );
+        await responder.start();
+
+        expect(responder.reEnqueueMissingItems([appointmentId])).to.eventually.be.rejectedWith(ArgumentError);
+
+        await responder.stop();
```

Comment:
Shouldn't we wait for reEnqueueMissingItems to complete, before calling `responder.stop()`?

Reply:
Updated